microcosm_eventsource/tests/fixtures.py

```diff
@@ -0,0 +1,130 @@
+"""
+Test fixtures.
+
+"""
+from six import add_metaclass
+
+from marshmallow import fields, Schema
+from microcosm.api import binding
+from microcosm_flask.fields import EnumField
+from microcosm_flask.namespaces import Namespace
+from microcosm_flask.session import register_session_factory
+from microcosm_postgres.context import SessionContext
+from microcosm_postgres.models import Model, UnixTimestampEntityMixin
+from microcosm_postgres.store import Store
+from sqlalchemy import Column, DateTime, String
+
+from microcosm_eventsource.controllers import EventController
+from microcosm_eventsource.event_types import all_of, but_not, info, EventType
+from microcosm_eventsource.models import EventMeta
+from microcosm_eventsource.resources import EventSchema, SearchEventSchema
+from microcosm_eventsource.routes import configure_event_crud
+from microcosm_eventsource.stores import EventStore
+
+
+class TaskEventType(EventType):
+    CREATED = info()
+    ASSIGNED = info(
+        follows=[all_of("CREATED", but_not("ASSIGNED"))],
+        accumulating=True,
+        requires=["assignee"]
+    )
+    SCHEDULED = info(
+        follows=[all_of("CREATED", but_not("SCHEDULED"))],
+        accumulating=True,
+        requires=["deadline"],
+    )
+    STARTED = info(
+        follows=[all_of("ASSIGNED", "SCHEDULED")],
+    )
+    REASSIGNED = info(
+        follows=["STARTED"],
+        accumulating=True,
+        requires=["assignee"],
+    )
+    RESCHEDULED = info(
+        follows=["STARTED"],
+        accumulating=True,
+        requires=["deadline"],
+    )
+    REVISED = info(
+        follows=["CREATED", "STARTED"],
+        # XXX need to support jumping to state CREATED
```

Comment:
Two choices here:
 - Let event types have more control over their state transition logic (probably)
 - Write the state machine to handle `REVISED` or `CREATED` everywhere.